### PR TITLE
#2799 Applies new last modified model to get a better lastmod date for articles.

### PR DESCRIPTION
**Pull request by ajrbyers**

### CODE REVIEW

File: src/core/model_utils.py [hunk 1]
```diff
@@ -358,3 +360,59 @@ def is_svg(f):
...
+                remote_field = manager.source_field_name
+                related_filter = {remote_field: self}
+                objects = manager.through.objects.filter(**related_filter)
+            elif field.one_to_many:
+                accessor = getattr(self, field.get_accessor_name())
+                objects = accessor.all()
+
+            for object in objects:
+                if hasattr(object, 'last_modified'):
+                    dates.append(getattr(object, 'last_modified'))
```
Comment (mauromsl) [inline src/core/model_utils.py:414]: I suspect this needs to be a recursive call, right? Or have an argument to make it optional at least.

**Review by mauromsl — CHANGES_REQUESTED** [review]
Looks good, I quite like the approach :)

One important comment regarding the calculation of the most recentd date_modified: I think this process needs to be recursive.
Also, we have no tests for the new behaviour

File: src/core/model_utils.py [hunk 2]
```diff
@@ -358,3 +360,59 @@ def is_svg(f):
...
+class AbstractLastModifiedModelManager(models.Manager):
+    def get_queryset(self):
+        # this is to use your custom queryset methods
+        return AbstractLastModifiedModelQuerySet(self.model, using=self._db)
+
+
+class AbstractLastModifiedModel(models.Model):
+    last_modified = models.DateTimeField(
+        auto_now=True,
+        auto_now_add=False,
```
Comment (mauromsl) [inline src/core/model_utils.py:380]: We should drop this as well, as want the date_modified to be populated on creation too.

**Comment by ajrbyers** [comment]
> Looks good, I quite like the approach :)

I'm glad you like the approach you recommended ;)

**Comment by ajrbyers** [comment]
@mauromsl can we carve some time tomorrow to take a look at the issues we discussed yesterday namely:

1. Recursive call
2. FKs not beeing looked up

**Pull request merged**